metric_map.go

```diff
@@ -176,6 +177,80 @@ func (mm *MetricMap) Split(count int) []*MetricMap {
 	return maps
 }
 
+func tagsMatch(tagNames []string, tagsKey string) string {
+	res := make([]string, 0)
+	for _, tv := range strings.Split(tagsKey, ",") {
+		for _, tagName := range tagNames {
+			if strings.HasPrefix(tv, tagName+":") {
+				res = append(res, tv)
+				break
+			}
+		}
+	}
+	return strings.Join(res, ",")
+}
+
+func (mm *MetricMap) SplitByTags(tagNames []string) map[string]*MetricMap {
+	maps := make(map[string]*MetricMap)
+	if len(tagNames) == 0 {
+		maps[""] = mm
```

Comment:
Should anything be added here if tagsNames is nil/empty?

Reply:
This shortcuts the case where dynamic-headers is not used. It makes the logic in `func (hfh *HttpForwarderHandlerV2) Run(ctx context.Context)` simpler. Otherwise, it'll need to duplicate the async post function call a bit, e.g.

```
if hfh.dynHeaders {
  for hdrs, mm := maps {
    // take lock
    // async post(mm, hdrs)
  }
} else {
  // take lock
  // async post(mm)
}
```

Reply:
Right.. That make sense :)